container-conf/playwright/tests/testing-utils.js

```diff
@@ -0,0 +1,279 @@
+    }
+    await page.waitForURL(`${HOST}/${applicationName}#/simulations`)
+}
+
+export let navigateToSimulation = async (page, simFolderNames) => {
+    for(var path of simFolderNames) {
+        //await page.waitForTimeout(1000);
+        await page.locator("div.sr-thumbnail span")
+        .locator(textFuzzyEquals(path)).click();
+        await page.waitForTimeout(500);
+    }
+}
+
+export let navigateToFirstSimulation = async (page) => {
+    let enterNextItem = async (page) => {
+        let firstItemLocator = page.locator(".sr-icon-col div.sr-thumbnail", { has: page.locator("span.glyphicon") }).first();
+        let elementHandle = await firstItemLocator.elementHandle();
+        let isFolder = await elementHandle.$eval("span.glyphicon", (node) => node.classList.contains("glyphicon-folder-close"));
+        await firstItemLocator.first().dblclick();
+        await page.waitForTimeout(250);
+        if(isFolder) {
+            await enterNextItem(page);
+        }
+    }
+
+    await enterNextItem(page);
+}
+
+export let openSimulationOptionsMenu = async(page) => {
+    await page.locator("a.dropdown-toggle > span.glyphicon.glyphicon-cog").click();
+}
+
+export let discardSimulationChanges = async(page) => {
+    await openSimulationOptionsMenu(page);
+    await page.locator(textFuzzyEquals("Discard Changes to Example")).click();
+    await page.locator("div.modal-dialog").locator(textFuzzyEquals("Discard Changes")).click();
+}
+
+export let findPlotByTitle = (page, title) => {
+    //return page.locator(`div[data-panel-title*="${title}"]`);
+    return page.locator("div.panel", {has: page.locator(textFuzzyEquals(title))});
+}
+
+export let waitForPlotToLoad = async (plotLocator, timeout) => {
+    await plotLocator.locator("css=div.sr-panel-loaded", {timeout}).click();
+    return plotLocator;
+}
+
+export let waitForPlotLoading = async (plotLocator, timeout) => {
+    await plotLocator.locator("div.sr-panel-loading", {timeout}).click();
+    return plotLocator;
+}
+
+export let getDownloadContents = (readable) => {
+    return new Promise((resolve, reject) => {
+        const chunks = [];
+
+        readable.on('readable', () => {
+            let chunk;
+            while (null !== (chunk = readable.read())) {
+                chunks.push(chunk);
+            }
+        });
+    
+        readable.on('end', () => {
+            const content = chunks.join('');
+            resolve(content);
+        });
+    })
+}
+
+export let openPanelOptions = async (panelLocator) => {
+    let panelDownloadMenuButtonLocator = panelLocator.locator(".sr-panel-options .glyphicon-cloud-download").first();
+    await panelDownloadMenuButtonLocator.click();
+    return panelDownloadMenuButtonLocator;
+}
+
+export let startDownload = async (page, downloadCausePromise) => {
+    let [ download ] = await Promise.all([
+        page.waitForEvent("download"),
+        downloadCausePromise
```

Comment:
Our style is comma terminator for list elements so that you don't get errors not including the comma on the previous line.